pkg/kubelet/cm/topologymanager/policy.go

```diff
@@ -271,15 +284,14 @@ func mergeFilteredHints(numaNodes []int, filteredHints [][]TopologyHint) Topolog
 	iterateAllProviderTopologyHints(filteredHints, func(permutation []TopologyHint) {
 		// Get the NUMANodeAffinity from each hint in the permutation and see if any
 		// of them encode unpreferred allocations.
-		mergedHint := mergePermutation(numaNodes, permutation)
+		mergedHint := mergePermutation(defaultAffinity, permutation)
```

Comment:
Is this change necessary for this new feature, or just a cleanup of sorts?
I don't see the advantage of doing it this way over the old way (unless I am missing something).

Reply:
For me it made code more readable because we passed `numaNodes` to `mergePemutation` only to calculate `defaultAffinity`, moreover we are removing additional calculation of `defaultAffinity` when `bestHint == nil`. I can revert this change if it makes the code more readable only for me.